pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -250,8 +250,9 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tp *v1alpha1.TektonPipel
 		return nil
 	} else if ready.Status == corev1.ConditionFalse {
```

Comment:
there is a short period when deployments are applied and pods are not yet running 
installerset sets ready status as false 
which TP reads that and sets its own status as false too
in that case would this logic consider that as `PreemptDeadlock` and delete resources?

Reply:
https://github.com/tektoncd/operator/blob/eb53caf21f0eb83f7820ccd42accfa29de5caf1e/pkg/reconciler/kubernetes/tektoninstallerset/install.go#L334
this has to be improved ..
where we need to differentiated false and waiting state

Reply:
`PreemptDeadlock` will only remove "rules" from Webhook, ie, set it to its initial state. As the webhookDefinition itself is not removed when the Deployment finally comes up, the rules will get populated and future edit operations on configMaps will be validated.

In addition, the `PreemptDeadlock` acts only if 1 or more configMaps are missing. So in majority of the time, if configMaps are there `PreemptDeadlock` does nothing. 

the deleteDeploymentCall in the `PreemptDeadlock` is a simple and effective way to trigger a reconcile loop in Installlerset making in recreate the configMaps.

Reply:
changed the mechanism by moving the focus to "Endpoint" resource rather than configMaps.